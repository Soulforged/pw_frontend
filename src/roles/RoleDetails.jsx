//@flow
import React from "react";
import { CRUDDetails, ActiveCell } from "src/components";

type Props = {
  id: number,
  item: Object,
  openForm: (number) => void,
  fetchRole: (number) => void,
};

const fields = {
  status: {
    component: ({ status }: { status: string }) => <ActiveCell value={status} />
  }
};

export default ({
  id, item, openForm, fetchRole
}: Props) => (
  <CRUDDetails
    id={id}
    item={item}
    openForm={openForm}
    loader={fetchRole}
    fields={fields}
    editButtonTitle="Edit Role"
  />
);
