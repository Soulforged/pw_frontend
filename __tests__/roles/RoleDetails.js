//@flow
import React from "react";
import RoleDetails from "src/roles/RoleDetails";
import renderer from "react-test-renderer";

it("details matches snapshot", () => {
  const rendered = renderer.create(<RoleDetails fetchRole={jest.fn()} />).toJSON();
  expect(rendered).toMatchSnapshot();
});

it("details matches snapshot with data", () => {
  const rendered = renderer.create(<RoleDetails fetchRole={jest.fn()} item={{ id: 1 }} />).toJSON();
  expect(rendered).toMatchSnapshot();
});
