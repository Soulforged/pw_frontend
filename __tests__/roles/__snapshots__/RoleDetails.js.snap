// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`details matches snapshot 1`] = `
<div
  className="details pnls"
  id="main-pnl"
>
  <div
    className="modal-p details"
  >
    Not found
  </div>
</div>
`;

exports[`details matches snapshot with data 1`] = `
<div
  className="details pnls"
  id="main-pnl"
>
  <div
    className="details"
  >
    <h5
      className="form-header trebuchet text-center bold"
    >
      Details for 
      1
      <button
        className="add-new-btn pointer pull-right bold"
        onClick={[Function]}
        title="Edit"
      >
        <i
          className="edit fa fa-pencil theme"
        />
        Edit Role
      </button>
    </h5>
    <div
      className="edit-pnl-cnt"
    >
      <div
        className="row"
      >
        <div
          className="col-md-6 col-sm-6"
        >
          Role name:
        </div>
        <div
          className="col-md-6 col-sm-6"
        />
      </div>
      <div
        className="row"
      >
        <div
          className="col-md-6 col-sm-6"
        >
          Status:
        </div>
        <div
          className="col-md-6 col-sm-6"
        >
          <span
            className="td-error"
          >
            <i
              className="fa fa-circle"
            />
            Inactive
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
